File: Java-Spring/src/main/java/org/codecop/lcdnumbers/DigitFactoryImpl.java
package org.codecop.lcdnumbers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 * Creates (LCD) digits from patterns.
 */
@Service
public class DigitFactoryImpl implements DigitFactory {

    private final Patterns patterns;

    public DigitFactoryImpl(Patterns patterns) {
        this.patterns = patterns;
    }

    @Override
    public Digit create(int digit) {
        return new Digit(digit, patterns);
    }
}

File: Kotlin-Dagger2/src/main/kotlin/org/codecop/lcdnumbers/DigitFactoryImpl.kt
package org.codecop.lcdnumbers

import javax.inject.Inject

/**
 * Creates (LCD) digits from patterns.
 */
class DigitFactoryImpl @Inject constructor(private val patterns: Patterns) : DigitFactory {
    override fun create(digit: Int) = Digit(digit, patterns)
}